src/uu/cp/src/cp.rs

```diff
@@ -1156,9 +1219,21 @@ fn copy_helper(source: &Path, dest: &Path, options: &Options) -> CopyResult<()>
                 ReflinkMode::Never => {}
             }
         }
+    } else if options.no_dereference && fs::symlink_metadata(&source)?.file_type().is_symlink() {
+        // Here, we will copy the symlink itself (actually, just recreate it)
+        let link = fs::read_link(&source)?;
+        let dest = if dest.is_dir() {
+            // the target is a directory, we need to keep the filename
+            let p = Path::new(source.file_name().unwrap());
+            dest.join(p)
+        } else {
+            dest.to_path_buf()
```

Comment:
To avoid this unnecessary allocation, this `if-else` should evaluate to `Cow<‘_, Path>`.